keras_hub/src/utils/openvino_utils.py

```diff
@@ -0,0 +1,311 @@
+import ast
+import functools
+from pathlib import Path
+
+from keras import tree
+
+from keras_hub.src.utils.keras_utils import print_msg
+
+try:
+    import openvino as ov
+    import openvino.opset14 as ov_opset
+    from openvino import Core
+
+    core = Core()
+except ImportError:
+    ov = None
+    ov_opset = None
+    core = None
+
+
+def load_openvino_supported_tools(config_file_path):
+    """Load OpenVINO supported models from whitelist file.
+
+    Args:
+        config_file_path: Path to whitelist file.
+
+    Returns:
+        list: Supported model paths.
+    """
+    try:
+        with open(config_file_path, "r") as f:
+            return [
+                line.strip()
+                for line in f
+                if line.strip() and not line.strip().startswith("#")
+            ]
+    except FileNotFoundError:
+        return []
+
+
+def setup_openvino_test_config(config_file_path):
+    """Setup OpenVINO test configuration with whitelist approach.
+
+    Args:
+        config_file_path: Path to the config file directory.
+
+    Returns:
+        list: Supported paths (whitelist) for OpenVINO testing.
+    """
+    return load_openvino_supported_tools(
+        Path(config_file_path) / "openvino_supported_tests.txt"
+    )
+
+
+@functools.lru_cache(maxsize=256)
+def _contains_training_methods(file_path, test_name):
```

Comment:
this feels still much too complex. i wonder if we can get away with something like this, in conftest.py only on the openvino backend

```python
keras.Model.compile = lambda *args, **kwargs: pytest.skip("todo msg")
keras.Model.fit = lambda *args, **kwargs: pytest.skip("todo msg")
```